core/sql/exp/ExpLOBaccess.cpp

```diff
@@ -552,8 +552,9 @@ Ex_Lob_Error ExLob::statSourceFile(char *srcfile, Int64 &sourceEOF)
    if (srcType == HDFS_FILE)
      {
        hdfsFile sourceFile = hdfsOpenFile(fs_,srcfile,O_RDONLY,0,0,0);   
```

Comment:
No not this. There are 3 other places in this file where it's set. But still experimenting with this fix. If this works, a new error number will be generated  right after the call to hdfsOpenFile . If errno is EAGAIN or ENOENT, the new error message will be  returned and we will retry based on that error at the higher layer.  But  the suggestion to return errno and perhaps also the filename as an output param may be a good one too. Will consider it.